tests/cli/test_compat.py

```diff
@@ -22,9 +41,52 @@ def test_no_stdout(monkeypatch: pytest.MonkeyPatch):
     assert module is not streamlink_cli.compat
 
     spec.loader.exec_module(module)
-    assert module.sys.stdout is None
-    assert isinstance(module.stdout, BufferedWriter)
-    assert module.stdout.name == devnull
 
-    assert mock_atexit_register.call_args_list == [call(module.stdout.close)]
-    module.stdout.close()  # close manually, since we've mocked the atexit.register() call
+    try:
+        yield module
+    finally:
+        # close manually, since we've mocked the atexit.register() call
```

Comment:
Another comment I don't think you need personally. The updated code is pretty clear but it's up to you.